hax/hax/motr/__init__.py

```diff
@@ -405,17 +429,48 @@ def notify_node_status(self,
 
         node_fid = self.consul_util.get_node_fid(node)
         encl_fid = self.consul_util.get_node_encl_fid(node)
-        ctrl_fid = self.consul_util.get_node_ctrl_fid(node)
-        LOG.debug('node_fid: %s encl_fid: %s ctrl_fid: %s with state: %s',
-                  node_fid, encl_fid, ctrl_fid, new_state)
+        ctrl_fids = self.consul_util.get_node_ctrl_fids(node)
+        LOG.debug('node_fid: %s encl_fid: %s ctrl_fids: %s with state: %s',
+                  node_fid, encl_fid, ctrl_fids, new_state)
 
         notes = []
-        if node_fid and encl_fid and ctrl_fid:
+        if node_fid and encl_fid:
             notes = [HaNoteStruct(no_id=x.to_c(), no_state=new_state)
-                     for x in [node_fid, encl_fid, ctrl_fid]]
+                     for x in [node_fid, encl_fid]]
+
+        if int(len(ctrl_fids)) != 0:
+            for x in ctrl_fids:
+                notes.append(HaNoteStruct(no_id=x.to_c(), no_state=new_state))
 
         return notes
 
+    def notify_ctrl_status(self,
+                           proc_note: HaNoteStruct) -> Optional[HaNoteStruct]:
+        new_state = proc_note.no_state
+        proc_fid = Fid.from_struct(proc_note.no_id)
+        assert ObjT.PROCESS.value == proc_fid.container
+        LOG.debug('Notifying ctrl status for process_fid=%s state=%s',
+                  proc_fid, new_state)
+
+        ctrl_fid = self.consul_util.get_ioservice_ctrl_fid(proc_fid)
+
+        if ctrl_fid is not None:
+            return HaNoteStruct(no_id=ctrl_fid.to_c(), no_state=new_state)
+        else:
+            return None
+
+    # Check if all the IO services of node are failed
+    def check_for_node_failure(self, proc_note: HaNoteStruct) -> bool:
```

Comment:
@mssawant comments:
1. can you change name of his function to "is_node_failed"
2. can we pass process fids instead of HaNoteStruct?

Reply:
`can you change name of his function to "is_node_failed"`
Done
'can we pass process fids instead of HaNoteStruct?'
We dont have process fids here, that info is present in consul which can be accesses by utils only